compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -145,6 +148,27 @@ kindOf = \case
   TStruct recordType -> checkRecordType recordType $> KStar
   TNat _ -> pure KNat
 
+expandTypeSynonyms :: MonadGamma m => Type -> m Type
+expandTypeSynonyms = expand where
+  expand = \case
+    TVar v -> return $ TVar v
+    TSyn tsyn -> do
```

Comment:
Given that we have the fully applied type synonym in the protobuf why don’t we preserve this in the Haskell AST and have a constructor of the form `TSynApp TypeSynName [Type]`. That makes it easier to check if it’s correctly applied.

Reply:
Yes. This is much nicer. Done